doc_src/index.hdr.in

```diff
@@ -17,7 +17,199 @@
+
+- <b>`ARGUMENTS`</b>: input to the command
+
+For external <b>commands</b> the executable is usually stored on file and the file path must be included in the variable `$PATH`, so that the file can be found: see <a href="#variables-special">Special Variables</a>.
+
+<b>Options</b> come in two forms: a short name, that is a hyphen with a single letter; or a long name, consisting of two hyphens with words connected by hyphens. Example: `-h` and `--help` are often used to print a help message. Options are a fixed set, described in the manual pages of the command, see <a href="#man-page">Manual Pages</a>
+
+<b>Arguments</b> are the arbitrary input part of a command: often it is a file or directory name, sometimes it is a string or a list.
+
+Example:
+
+\fish
+>echo -s Hallo World!
+HalloWorld!
+\endfish
+
+-  both `Hallo` and `World!` are arguments to the echo command
+- `-s` is an option that suppresses spaces in the output of the command
+
+\subsection shell-programming Commands versus Programs
+
+<b>Programs</b> in other languages can often be regarded as black boxes: they get complex input and return complex output. Sometimes they produce side effects such as writing to a file or reporting an error, but the emphasis is on: arguments in and return values out: 
+
+Arguments &rarr; |Program| &rarr; Return Values
+
+<b>Shell commands</b> are different:
+
+- the side effects take the center of the stage: they transform an <b>input stream of data</b> into an <b>output stream of data</b>. Both of these streams are usually the terminal, but they can be redirected.
+- the arguments become options or switches paired with data: the switches influence the behaviour of the command
+- the return value shrinks to an <b>exit code</b>: this exit code is 0 when the command executes normally and between 1 and 255 otherwise.
+
+<table with=100%>
+<tr><td>Input Stream &rArr;<td rowspan="2">|Shell Command|<td>&rArr; Output Stream
+<tr><td>switch and data as arguments &rarr;<td>&rarr; exit code
+</table>
+
+This leads to another way of programming and especially of combining commands:
+
+There are two ways to combine shell commands:
+
+- Commands can pass on their streams to each other: one command takes the output stream of another command as its input stream and the two commands execute in parallel. This is called piping, see <a href="#piping">Piping</a>.
+
+Example:
+
+\fish
+# Every line of the `ls` command is immediatelly passed on to the `grep` command
+>ls -l | grep "my topic"
+\endfish
+
+- Commands can pass on all their output as a chunk</b>: the output stream of one command is bundled and taken as data argument for the second command. This is called command substitution, see <a href="#expand-command-substitution">Command Substitution</a>.
+
+Example:
+\fish
+# the output of the inner `ls` command is taken as the input argument for the outer `echo` command
+>echo (ls a*)
+\endfish
+
+\subsection shebang-line Shebang Line
```

Comment:
I think this is a useful concept - a shell script as a program - but perhaps could be introduced later on. My sense is that new users are focused on interactive mode. Would it be better to have a section called "interactive use and script use" where shebangs are included?

Reply:
Sounds good.